flash/text/seq2seq/question_answering/model.py

```diff
@@ -70,11 +71,14 @@ def __init__(
             rouge_newline_sep=rouge_newline_sep,
             use_stemmer=use_stemmer,
         )
+        self.squad_v2 = load_metric("squad_v2")
```

Comment:
We should `hard-code` anything related to a `dataset` within a model. You could use state to propagate the metric or dataset name to the model directly.
If you need more explication there, please find us on the Gather :)

Reply:
That's just a metric though right? Can presumably be used with any data set?

Reply:
I don't think it could. But even then, we shouldn't force users to use the metrics associated to `squad_v2` dataset if they don't want to.

Maybe, let's just to  `metrics: Union[Metric, Callable, Mapping, Sequence, None] = load_metric("squad_v2")`